src/QMCHamiltonians/BareKineticEnergy.h

```diff
@@ -138,6 +165,30 @@ class BareKineticEnergy : public OperatorBase
   // Nothing is done on GPU here, just copy into vector
   void addEnergy(MCWalkerConfiguration& W, std::vector<RealType>& LocalEnergy) override;
 #endif
+  /** initialize a shared resource and hand it to a collection
+   */
+  void createResource(ResourceCollection& collection) const override;
+
+  /** acquire a shared resource from a collection
+   */
+  void acquireResource(ResourceCollection& collection, const RefVectorWithLeader<OperatorBase>& o_list) const override;
+
+  /** return a shared resource to a collection
+   */
+  void releaseResource(ResourceCollection& collection, const RefVectorWithLeader<OperatorBase>& o_list) const override;
+
+private:
+  struct MultiWalkerResource : public Resource
```

Comment:
I think we can keep `struct MultiWalkerResource;` in the header and move the definition to cpp.

Regarding the name. I used to define the resource struct outside the resource owner class, then carrying the owner class name was necessary. However here, the resource class is defined inside the owner class scope. So not carrying the owner class name should be fine.

Reply:
I have a good number of PR's coming I'll look at how this usage works for them and make it consistent along the way.  I think @markdewing has a point.  Will leave as is for now.